pkg/controller/node/rate_limited_queue.go

```diff
@@ -0,0 +1,177 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodecontroller
+
+import (
+	"container/heap"
+	"sync"
+	"time"
+
+	"k8s.io/kubernetes/pkg/util"
+)
+
+// TimedValue is a value that should be processed at a designated time.
+type TimedValue struct {
+	Value string
+	Added time.Time
+	Next  time.Time
```

Comment:
I'd like names to be more descriptive. 'Added' is probably good enough (though I'd prefer 'AddedAt', 'AddTime' or sth), but 'Next' does not say anything. In this context probably 'ProcessAt' would be better. @bgrant0607 can probably come up with better names.